src/xmessage_builder.cpp

```diff
@@ -0,0 +1,106 @@
+#include <string>
+#include <vector>
+
+#include "nlohmann/json.hpp"
+
+#include "xeus/xmessage_builder.hpp"
+
+namespace nl = nlohmann;
+
+namespace xeus
+{
+    nl::json create_error_reply(const std::string& evalue,
+                                const std::string& ename,
+                                const std::vector<std::string>& trace_back)
+    {
+        nl::json kernel_res;
+        kernel_res["status"] = "error";
+        kernel_res["ename"] = ename;
+        kernel_res["evalue"] = evalue;
+        kernel_res["traceback"] = trace_back;
+        return kernel_res;
+    }
+
+    nl::json create_successful_reply(const nl::json& data,
+                                     const nl::json& payload,
+                                     const nl::json& user_expressions)
+    {
+        nl::json kernel_res;
+        kernel_res["status"] = "ok";
+        kernel_res = data;
+        kernel_res["payload"] = payload;
+        kernel_res["user_expressions"] = user_expressions;
+        return kernel_res;
+    }
+
+    nl::json create_complete_reply(const std::vector<std::string>& matches,
+                                   const int& cursor_start,
+                                   const int& cursor_end,
+                                   const std::string& code,
+                                   const nl::json& metadata)
+    {
+        nl::json kernel_res;
+        kernel_res["status"] = "ok";
+        kernel_res["code"] = code;
+        kernel_res["matches"] = matches;
+        kernel_res["cursor_start"] = cursor_start;
+        kernel_res["cursor_end"] = cursor_end;
+        kernel_res["metadata"] = metadata;
+        return kernel_res;
+    }
+
+    nl::json create_inspect_reply(const nl::json& data,
```

Comment:
It might make sense to have `found` as first argument, with this signature:
```cpp
nl::json create_inspect_reply(const bool& found,
                              const nl::json& data = nl::json::object(),
                              const nl::json& metadata = nl::json::object())
```

I mean that `metadata` will almost never be provided, so it's more convenient to call:
```cpp
create_inspect_reply(true, data);
```

instead of
```cpp
create_inspect_reply(data, nl::json::object(), true);
```

Reply:
nitcpicking: the bool argument should be passed by value.